### PR TITLE
change the step function

### PR DESCRIPTION
**Pull request by youngjaean**
offset is offset=sigmoid(delta/tau)+0.5.offset Conditional Statements  is None  but used value so not match the paper.

And not used delta. changed from rho * tens[~target & pos_mask & margin_mask ] to rho * tens[~target & pos_mask & margin_mask - delta]

### CODE REVIEW

File: roadmap/losses/smooth_rank_ap.py [hunk 1]
```diff
@@ -46,9 +46,9 @@ def step_rank(tens, tau, rho, offset, delta=None, start=0.5, target=None, genera
     else:
         margin_mask = tens > delta
         tens[~target & pos_mask & ~margin_mask] = start + tau_sigmoid(tens[~target & pos_mask & ~margin_mask], tau_p).type(tens.dtype)
-        if offset is None:
+        if offset is None: 
             offset = tau_sigmoid(torch.tensor([delta], device=tens.device), tau_p).type(tens.dtype) + start
-        tens[~target & pos_mask & margin_mask] = rho * tens[~target & pos_mask & margin_mask] + offset
+        tens[~target & pos_mask & margin_mask] = rho * tens[~target & pos_mask & margin_mask - delta] + offset
```
Comment (elias-ramzi) [inline roadmap/losses/smooth_rank_ap.py:51]: this should be `tens[~target & pos_mask & margin_mask] = rho *(tens[~target & pos_mask & margin_mask] - delta) + offset`

Comment (youngjaean) [reply roadmap/losses/smooth_rank_ap.py:51]: oh, i miss take. thx

**Comment by youngjaean** [comment]
I chage the not correct code!

**Comment by elias-ramzi** [comment]
I will run some tests to see that all is good and merge this, thanks!

**Review by elias-ramzi — APPROVED** [review]
This PR corrects the implementation of the `step_rank.py`function which corresponds to EQ. (4) of the paper.

**Comment by elias-ramzi** [comment]
Thank you @youngjaean for the PR, it has been merged!
